src/components/tabController/TabPage.tsx

```diff
@@ -49,18 +49,29 @@ export default function TabPage({
   }, [lazy, shouldLoad]);
 
   useAnimatedReaction(() => {
-    return targetPage.value === index;
+    return currentPage.value === index;
   },
   (isActive, wasActive) => {
     if (isActive) {
       runOnJS(lazyLoad)();
       runOnJS(setFocused)(true);
     }
-
-    if (wasActive && wasActive !== isActive) {
+    if (wasActive) {
       runOnJS(setFocused)(false);
     }
-  });
+  },
+  [currentPage]);
+
+  /* Handle freeze for pages that near the active page (relevant for carousel page) */
+  useAnimatedReaction(() => {
+    return asCarousel && currentPage.value - 1 === index || currentPage.value + 1 === index;
```

Comment:
I think we're missing parentheses:
```
return asCarousel && (currentPage.value - 1 === index || currentPage.value + 1 === index);
```

Reply:
Fixed that

Reply:
I think I forgot to delete this comment when I sent the review